core/src/main/scala/akka/kafka/internal/SubSourceLogic.scala

```diff
@@ -143,9 +146,12 @@ private[kafka] abstract class SubSourceLogic[K, V, Msg](
       }
   }
 
-  val partitionRevokedCB = getAsyncCallback[Set[TopicPartition]] { tps =>
+  val partitionRevokedCB = getAsyncCallback[Set[TopicPartition]] { revoked =>
+    subscription.rebalanceListener.foreach {
+      _.tell(TopicPartitionsRevoked(subscription, revoked), sourceActor.ref)
+    }
```

Comment:
I think we should keep these consistent: either send those messages from the callback, or from the client thread.

Reply:
You're so right, what was I thinking...